### PR TITLE
TRUS-1048

### PR DESCRIPTION
**Pull request by codebysachin**

### CODE REVIEW

File: app/models/AgentUKAddress.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json._
+
+case class AgentUKAddress (field1: String, field2: String)
```
Comment (adamconder) [inline app/models/AgentUKAddress.scala]: Delete this file.

File: app/forms/AgentUKAddressFormProvider.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+package forms
+
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+import play.api.data.Forms._
+import models.AgentUKAddress
+
+class AgentUKAddressFormProvider @Inject() extends Mappings {
```
Comment (adamconder) [inline app/forms/AgentUKAddressFormProvider.scala]: Delete this file.

File: test/forms/AgentUKAddressFormProviderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+package forms
+
+import forms.behaviours.StringFieldBehaviours
+import play.api.data.FormError
+
+class AgentUKAddressFormProviderSpec extends StringFieldBehaviours {
+
+  val form = new AgentUKAddressFormProvider()()
```
Comment (adamconder) [inline test/forms/AgentUKAddressFormProviderSpec.scala]: Delete this file.

File: test/generators/ModelGenerators.scala [hunk 4]
```diff
@@ -22,6 +22,14 @@ import org.scalacheck.{Arbitrary, Gen}
 
 trait ModelGenerators {
 
+  implicit lazy val arbitraryAgentUKAddress: Arbitrary[AgentUKAddress] =
+    Arbitrary {
+      for {
+        field1 <- arbitrary[String]
```
Comment (adamconder) [inline test/generators/ModelGenerators.scala]: Remove this.

File: test/generators/UserAnswersEntryGenerators.scala [hunk 5]
```diff
@@ -24,6 +24,14 @@ import play.api.libs.json.{JsValue, Json}
 
 trait UserAnswersEntryGenerators extends PageGenerators with ModelGenerators {
 
+  implicit lazy val arbitraryAgentUKAddressUserAnswersEntry: Arbitrary[(AgentUKAddressPage.type, JsValue)] =
+    Arbitrary {
+      for {
+        page  <- arbitrary[AgentUKAddressPage.type]
+        value <- arbitrary[AgentUKAddress].map(Json.toJson(_))
```
Comment (adamconder) [inline test/generators/UserAnswersEntryGenerators.scala]: Change this to be a UKAddress

File: app/models/AgentUKAddress.scala (hunk 1 above)
Comment (codebysachin) [reply app/models/AgentUKAddress.scala]: -resolve

File: app/forms/AgentUKAddressFormProvider.scala (hunk 2 above)
Comment (codebysachin) [reply app/forms/AgentUKAddressFormProvider.scala]: deleted

File: test/forms/AgentUKAddressFormProviderSpec.scala (hunk 3 above)
Comment (codebysachin) [reply test/forms/AgentUKAddressFormProviderSpec.scala]: deleted

File: test/generators/ModelGenerators.scala (hunk 4 above)
Comment (codebysachin) [reply test/generators/ModelGenerators.scala]: removed

File: test/generators/UserAnswersEntryGenerators.scala (hunk 5 above)
Comment (codebysachin) [reply test/generators/UserAnswersEntryGenerators.scala]: changed

**Pull request merged**